src/ripple/app/ledger/impl/DisputedTx.h

```diff
@@ -36,22 +36,23 @@ namespace ripple {
 
     Undisputed transactions have no corresponding @ref DisputedTx object.
 */
+
+template <class Tx_t, class TxID_t, class NodeID_t>
 class DisputedTx
 {
 public:
-    // VFALCO `Blob` is a poor choice of parameter
-    DisputedTx (uint256 const& txID,
-            Blob const& tx, bool ourVote, beast::Journal j)
-        : mTransactionID (txID)
+    DisputedTx (Tx_t const& tx,
+            bool ourVote, beast::Journal j)
+        : mTransactionID (tx.getID())
         , mYays (0)
         , mNays (0)
         , mOurVote (ourVote)
-        , transaction (tx.data(), tx.size())
+        , transaction (tx)
         , j_ (j)
     {
     }
 
-    uint256 const& getTransactionID () const
+    TxID_t const& getTransactionID () const
```

Comment:
I would have gone with `tid` or `txid` instead of `getTransactionID`. Maybe I'm getting old and lazy, who knows.